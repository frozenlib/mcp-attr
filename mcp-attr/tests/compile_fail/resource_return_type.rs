use mcp_attr::{
    server::{mcp_server, McpServer},
    Result,
};
struct X;

struct MyServer;

#[mcp_server]
impl McpServer for MyServer {
    #[resource]
    async fn f(&self) -> Result<X> {
        todo!()
    }
}

fn main() {}
