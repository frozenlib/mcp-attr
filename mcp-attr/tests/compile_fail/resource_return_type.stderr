error[E0277]: the trait bound `mcp_attr::schema::ReadResourceResult: From<X>` is not satisfied
  --> tests/compile_fail/resource_return_type.rs:12:26
   |
12 |     async fn f(&self) -> Result<X> {
   |                          ^^^^^^ the trait `From<X>` is not implemented for `mcp_attr::schema::ReadResourceResult`
   |
   = help: the following other types implement trait `From<T>`:
             `mcp_attr::schema::ReadResourceResult` implements `From<&ReadResourceResultContentsItem>`
             `mcp_attr::schema::ReadResourceResult` implements `From<&mcp_attr::schema::BlobResourceContents>`
             `mcp_attr::schema::ReadResourceResult` implements `From<&mcp_attr::schema::ReadResourceResult>`
             `mcp_attr::schema::ReadResourceResult` implements `From<&mcp_attr::schema::TextResourceContents>`
             `mcp_attr::schema::ReadResourceResult` implements `From<&str>`
             `mcp_attr::schema::ReadResourceResult` implements `From<Base64Bytes>`
             `mcp_attr::schema::ReadResourceResult` implements `From<ReadResourceResultContentsItem>`
             `mcp_attr::schema::ReadResourceResult` implements `From<Vec<ReadResourceResultContentsItem>>`
           and $N others
